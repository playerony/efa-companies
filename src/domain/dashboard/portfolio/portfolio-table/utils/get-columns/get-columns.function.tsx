import { MouseEvent } from 'react';
import { TableColumnProps } from 'antd';

import { Label } from '@ui';

import { Company } from '../../../../types';
import { Attributes } from './get-columns.types';

const baseColumnsDefinitions = [
  {
    dataIndex: 'name',
    title: 'Company Name',
  },
  {
    title: 'Symbol',
    dataIndex: 'symbol',
  },
];

export const getColumns = ({ onRemove }: Attributes): TableColumnProps<Company>[] => {
  const handleRemoveClick = (record: Company) => (event: MouseEvent) => {
    event.stopPropagation();

    onRemove(record.symbol);
  };

  const actionsColumnRender = (_: unknown, record: Company) => (
    <Label strong pointer onClick={handleRemoveClick(record)}>
      Remove
    </Label>
  );

  return [
    ...baseColumnsDefinitions,
    {
      title: 'Actions',
      render: actionsColumnRender,
    },
  ];
};
