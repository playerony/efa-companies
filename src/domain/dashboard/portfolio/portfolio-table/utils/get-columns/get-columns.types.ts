export interface Attributes {
  onRemove: (symbol: string) => void;
}
