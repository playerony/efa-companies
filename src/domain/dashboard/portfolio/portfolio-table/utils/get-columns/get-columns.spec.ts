import { getColumns } from './get-columns.function';

import { functionImportTest } from '@utils';

describe('getColumns Function', () => {
  functionImportTest(getColumns);

  it('should return an array which represents each table column', () => {
    const onRemove = jest.fn();
    const columns = getColumns({ onRemove });

    expect(columns).toMatchSnapshot();
  });
});
