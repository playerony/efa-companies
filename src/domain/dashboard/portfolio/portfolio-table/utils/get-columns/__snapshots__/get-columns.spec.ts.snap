// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getColumns Function should return an array which represents each table column 1`] = `
Array [
  Object {
    "dataIndex": "name",
    "title": "Company Name",
  },
  Object {
    "dataIndex": "symbol",
    "title": "Symbol",
  },
  Object {
    "render": [Function],
    "title": "Actions",
  },
]
`;
